src/tss2-tcti/tcti-device.c

```diff
@@ -393,7 +421,11 @@ Tss2_Tcti_Device_Init (
     memset (&tcti_common->header, 0, sizeof (tcti_common->header));
     tcti_common->locality = 3;
 
+#ifdef __VXWORKS__
+    tcti_dev->fd = open (dev_path, O_RDWR, (S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP));
```

Comment:
What's the function of the mode, if O_CREAT is not set ?
IMO, it is being ignored, right ?
If that is the case, then just write `0 /* ignored */` instead of these values.

Reply:
For file system, mode parameter is only useful when set O_CREAT. For device, it depends on the implementation. TPM driver of VxWorks will check the mode always, but Linux driver doesn’t. Therefore, it works on Linux without mode parameter, but doesn’t work on VxWorks.
The code is more clean if removed the ifdef, but maybe the value of the mode parameter doesn’t satisfy all of the OS. It will introduce other ifdef.